src/components/MessagesList/MessagesGroup/Message/MessagePart/FilePreview.vue

```diff
@@ -152,26 +160,46 @@ export default {
 		defaultIconUrl() {
 			return imagePath('core', 'filetypes/file')
 		},
-		previewSizeClass() {
+		previewImageClass() {
+			let classes = ''
 			if (this.previewSize === 64) {
-				return 'preview-64'
+				classes += 'preview-64 '
+			} else {
+				classes += 'preview '
+			}
+
+			if (this.previewType === PREVIEW_TYPE.MIME_ICON) {
+				classes += 'mimeicon'
 			}
-			return 'preview'
+			return classes
 		},
-		previewUrl() {
-			const userId = this.$store.getters.getUserId()
+		previewType() {
 			if (this.hasTemporaryImageUrl) {
-				return this.localUrl
+				return PREVIEW_TYPE.TEMPORARY
 			}
 
 			if (this.previewAvailable !== 'yes') {
-				return OC.MimeType.getIconUrl(this.mimetype)
+				return PREVIEW_TYPE.MIME_ICON
 			}
 
-			// max size of a gif for which we allow direct embedding
 			const maxGifSize = getCapabilities()?.caps?.spreed?.config?.previews?.['max-gif-size'] || 3145728
 			if (this.mimetype === 'image/gif' && this.size <= maxGifSize) {
-				// return direct image so it can be animated
+				return PREVIEW_TYPE.DIRECT
+			}
+
+			return PREVIEW_TYPE.PREVIEW
+		},
+		previewUrl() {
+			const userId = this.$store.getters.getUserId()
+
+			switch (this.previewType) {
```

Comment:
I think that this would read better with ifs, like:
```
if (this.previewType === PREVIEW_TYPE.TEMPORARY) {
    return this.localUrl
}

if (this.previewType === PREVIEW_TYPE.MIME_ICON) {
    return OC.MimeType.getIconUrl(this.mimetype)
}

if (this.previewType === PREVIEW_TYPE.DIRECT && userId === null) {
    ...
```

But I guess it is just a matter of taste :-)

Reply:
in hindsight I also prefer the if version, will adjust